content/en/blog/_posts/2023-09-15-Bootstrap-an-Air-Gapped-Cluster-with-Kubeadm/index.md

```diff
@@ -0,0 +1,665 @@
+- service systemd file
+- kubeadm configuration file
+- Docker registry container image
+- Kubernetes component container images
+- CNI network plugin container images ([Flannel](https://github.com/flannel-io/flannel) will be used for this lab)
+- CNI network plugin manifests
+- CNI tooling container images
+
+The way I identified these was by trying to do the installation and working through all of the errors that are thrown around an additional dependency being required. In a real air-gapped scenario, each transport of artifacts across the air gap could represent anywhere from 20 minutes to several weeks of time spent by the installer. That is to say that the target system could be located in a data center on the same floor as your desk, at a satellite downlink facility in the middle of nowhere, or on a submarine that's out to sea. Knowing what is on that system at any given time is important so you know what you have to bring.
+
+# Prepare the Node for K8s
+
+Before downloading and moving the artifacts to the VM, let's first prep that VM to run Kubernetes.
+
+### As a normal user:
+
+**Make destination directory for software artifacts**
+
+```bash
+mkdir ~/tmp
+```
+### As root:
+
+**Write to /etc/sysctl.d/99-k8s-cri.conf**
+
+```bash
+cat > /etc/sysctl.d/99-k8s-cri.conf << EOF
+net.bridge.bridge-nf-call-iptables=1
+net.ipv4.ip_forward=1
+net.bridge.bridge-nf-call-ip6tables=1
+EOF
+```
+
+**Write to /etc/modules-load.d/k8s.conf**
+```bash
+echo -e overlay\\nbr_netfilter > /etc/modules-load.d/k8s.conf
+```
+**Install iptables-legacy**
+```bash
+dnf -y install iptables-legacy
+```
+**Set iptables to use iptables-legacy**
+```bash
+update-alternatives --set iptables /usr/sbin/iptables-legacy
+```
+**Turn off swap**
+```bash
+touch /etc/systemd/zram-generator.conf
+systemctl mask systemd-zram-setup@.service
+sed -i '/ swap / s/^\(.*\)$/#\1/g' /etc/fstab
+```
+
+**Disable firewalld**
+```bash
+systemctl disable --now firewalld
+```
+**Disable systemd-resolved**
+```bash
+systemctl disable --now systemd-resolved
+```
+**Configure DNS defaults for NetworkManager**
+```bash
+sed -i '/\[main\]/a dns=default' /etc/NetworkManager/NetworkManager.conf
+```
+**Blank the system-level DNS resolver configuration**
+
+```bash
+unlink /etc/resolv.conf || true
+touch /etc/resolv.conf
+```
+
+**Disable SELinux (just for a demo - check before doing this in production!)**
+
+```bash
+setenforce 0
+```
+
+**Make sure all changes survive a reboot**
+```bash
+reboot
+```
+# Download all the artifacts
+
+On the laptop/host machine, download all of the artifacts enumerated in the previous section.  Since the air gapped VM is running Fedora 37, all of the dependencies shown in this part are for Fedora 37.  Note, this procedure will only work on AArch64 or AMD64 CPU architectures.  Execute this procedure anywhere where you have write permission, your home directory is perfectly suitable.
```

Comment:
This sentence`Execute this procedure anywhere where you have write permission, your home directory is perfectly suitable.` can be changed to 
`You can execute this procedure anywhere you have write permissions; your home directory is a perfectly suitable choice.`
This looks more user friendly.

Reply:
@Vikash-8090-Yadav , agreed. Your suggestion reads a lot friendlier.  Thanks!